Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs

```diff
@@ -216,28 +237,78 @@ protected async Task<string> PerformRequestAsync(HttpMethod httpMethod, string p
-            if((bool)formUrlEncodedContent?.Any())
+            if ((bool)formUrlEncodedContent?.Any())
             {
                 content = new FormUrlEncodedContent(formUrlEncodedContent.ToList());
             }
+
             if (!path.StartsWith("/"))
             {
                 path = $"/{path}";
             }
 
-            var request = new HttpRequestMessage(httpMethod, $"{GetBasePath()}{path}")
+            var request = GetHttpRequestMessage(httpMethod, $"{GetBasePath()}{path}", headers);
+            request.Content = content;
+
+            HttpResponseMessage response = await client.SendAsync(request).ConfigureAwait(false);
+            LastApiResponse = response;
+            return await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+        }
+
+        protected virtual async Task<string> PerformAuthorizationServerRequestAsync(HttpMethod httpMethod, string path, Dictionary<string, string> headers, string authorizationServerId = "default", params KeyValuePair<string, string>[] formUrlEncodedContent)
+        {
+            FormUrlEncodedContent content = null;
+            if ((bool)formUrlEncodedContent?.Any())
             {
-                Content = content
-            };
+                content = new FormUrlEncodedContent(formUrlEncodedContent.ToList());
+            }
+
+            if (!path.StartsWith("/"))
+            {
+                path = $"/{path}";
+            }
+
+            var request = GetHttpRequestMessage(httpMethod, $"{GetAuthorizationServerBasePath(authorizationServerId)}{path}", headers);
+            request.Content = content;
+
             HttpResponseMessage response = await client.SendAsync(request).ConfigureAwait(false);
             LastApiResponse = response;
-            string responseText = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-            return responseText;
+            return await response.Content.ReadAsStringAsync().ConfigureAwait(false);
         }
 
         protected virtual string GetBasePath()
         {
             return $"{Config?.OktaDomain}/oauth2/v1";
         }
+
+        protected virtual string GetAuthorizationServerBasePath(string authorizationServerId = "default")
+        {
+            return $"{Config?.OktaDomain}/oauth2/{authorizationServerId}/v1";
+        }
+
+        /// <summary>
+        /// Gets a request message.
+        /// </summary>
+        /// <param name="apiToken">The API token.</param>
+        /// <param name="method">The http method.</param>
+        /// <param name="path">The path.</param>
+        /// <returns>HttpRequestMessage</returns>
+        protected HttpRequestMessage GetHttpRequestMessage(HttpMethod method, string path, Dictionary<string, string> headers)
+        {
+            HttpRequestMessage request = new HttpRequestMessage(method, path);
+            request.Headers.Add("Accept", "application/json");
+            request.Headers.Add("User-Agent", OktaXamarinUserAgent.Value);
```

Comment:
Just noticed this is getting added twice.  I'll fix in upcoming PR for `introspect` implementation.